3 feature 3 create and customize workout routines

TimmyTurner691 opened the pull request:
This pull request will implement the functionality for feature 3. The design pattern that I chose to follow was the MVC design pattern, where I have my 'model' classes stored in management, my 'view' classes stored in the views package, and my 'controller' classes stored in the controllers package. When this feature is run, a view window will be displayed, and the controller class associated with it will load the data onto the view, this data is taken from the model, and the controller links it to the view. This follows the MVC pattern pretty well since both the view and the model DO NOT know of each other's existence and never interact with each other directly, and the controller is the one that takes both of them and links them together.

Commit:
edited hello world, just practicing pull request stuff.

Commit:
Finished all the methods we planned on using for the RoutineManager, RoutineHistory and Workout classes. The view is yet to be finished but, I prefer planning it out in person so that way we can coordinate easier. The code for the Workout class can't run yet since it depends on having class Exercise implemented.

Commit:
Added an incomplete interface for the workouts display page. I also seperated all my classes into Model, View and Controller folders

bdls-jamal reviewed (changes requested):
Coding Style:
- general structure is solid and readable
- helpful to have in line comments to assist with understanding what the code is doing
- 'UpdateHistory' has first letter capitalization, different from other methods in the same class(if this is intentional, let us know the reasoning and then it is okay)
- 'CompleteWorkout', 'RemoveExercise', 'AddExercise' same as above

Commit:
Corrected all styling errors pointed out by @bdls-jamal, added javadocs to all classes in RoutineManagerModel, javadocs will be added to the view once it is completed

Commit:
Added two Controller classes allowing the view to display the exercises in each workout, I also made the View classes a lot cleaner as well and moved all actionlistener classes to the controllers.

jhalaksaraogi commented on src/main/java/RoutineManagerController/WorkoutMenuController.java:
```diff
@@ -0,0 +1,35 @@
+
+        this.theView = theView;
+
+        this.theView.addWorkoutSelectionDisplay(new WorkoutSelectionDisplay());
```
Since this constructor does have a few parameters, it might be a good idea to add Javadoc indicating what each parameter is.

jhalaksaraogi commented on src/main/java/RoutineManagerModel/Workout.java:
```diff
@@ -0,0 +1,61 @@
+     */
+    public ArrayList<String> getWorkout() {
+        return this.workout;
+    }
```
Great work on the Javadoc for the RoutineManagerModel package. The purpose of each of the three classes is clear. Each method's javadoc follows conventions and is very detailed.

jhalaksaraogi commented on src/main/java/RoutineManagerController/WorkoutSelectionController.java:
```diff
@@ -0,0 +1,92 @@
+        }
+    }
+}
+
```
Pl add Javadoc to the methods and the classes in the Controller package.

jhalaksaraogi commented on src/main/java/RoutineManagerView/WorkoutSelection.java:
```diff
@@ -0,0 +1,58 @@
+        workout5Button.addActionListener(displayListener);
+    }
+
+}
```
Pl add Javadoc to the methods and the classes in the View package.

Commit:
You can now remove an exercise from a workout routine. The code is very messy, I'm going to create a separate class to create the view for the workout menu that way, there is no duplicate code.

Commit:
Removed all duplicate code in WorkoutMenuController, WorkoutSelectionController, and RemoveExerciseMenuController by implementing two interfaces;
- DisplayExercisesWorkoutMenu
- DisplayExercisesRemoveMenu
These interfaces contain methods that allow WorkoutMenuController and RemoveExerciseMenuController to display exercises from each workout in the UI.

Commit:
You are now able to add exercises from a list of exercises I added temporarily until I get the Exercise class. With this I finished everything I was able to implement with the classes I have and now the feature is fully functional for now.

bdls-jamal reviewed (dismissed):
Code Styling:
- Changes were made to the notes in a previous comment.
- code is structured well and is readible

absolute-xero7 reviewed (dismissed):
The general coding structure is good.
Clean architecture is being followed.
Can use more javadoc to make the code easy to follow.

Commit:
Added more JavaDoc to my View classes.

TimmyTurner691 commented:
This Pull Request was merged to the main, however, the JavaDoc review is yet to be finished. We decided to merge right away to test our functionality. The JavaDoc will be finished TODAY.

Commit:
Added more JavaDoc to my View classes.

AHui156 reviewed (commented):
No testings have been done yet
